pkg/raft/fsm/fsm.go

```diff
@@ -0,0 +1,102 @@
+package fsm
+
+import (
+	"bufio"
+	"encoding/base64"
+	"io"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/hashicorp/raft"
+	"github.com/rs/zerolog/log"
+)
+
+type CommunityStateMachine struct {
+	Log       []string `json:"log"`
+	StateJSON []byte
+}
+
+func NewCommunityStateMachine() *CommunityStateMachine {
+	return &CommunityStateMachine{
+		Log:       make([]string, 0),
+		StateJSON: []byte("{}"),
+	}
+}
+
+// Apply log is invoked once a log entry is committed.
```

Comment:
this comment is really nice. can you add some like this to the raft/community library before major functions for clarity?

Reply:
Lmao I didn't write these comments, they were copied from Raft. I'll get good comments up on the other stuff. Also I forgot that I need to finish some READMEs/documentation